zerver/lib/onboarding.py

```diff
@@ -187,7 +187,25 @@ def send_welcome_bot_response(send_request: SendMessageRequest) -> None:
     to welcome-bot, trigger the welcome-bot reply."""
     welcome_bot = get_system_bot(settings.WELCOME_BOT, send_request.realm.id)
     human_response_lower = send_request.message.content.lower()
+    human_user_recipient_id = send_request.message.sender.recipient_id
+    assert human_user_recipient_id is not None
     content = select_welcome_bot_response(human_response_lower)
+    realm_id = send_request.realm.id
+    commands = bot_commands()
+    if (
+        commands in content
+        and Message.objects.filter(
+            realm_id=realm_id,
+            sender_id=welcome_bot.id,
+            recipient_id=human_user_recipient_id,
+            content__icontains=commands,
+        ).exists()
+        # Uses index 'zerver_message_realm_sender_recipient'
```

Comment:
I checked after populating database with 3000 messages.. It uses 
```py
                                                                                             QUERY PLAN                                                                                             
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Limit  (cost=0.28..4.31 rows=1 width=4) (actual time=0.068..0.069 rows=0 loops=1)
   ->  Index Scan using zerver_message_realm_id_849a39c8 on zerver_message  (cost=0.28..4.31 rows=1 width=4) (actual time=0.065..0.066 rows=0 loops=1)
         Index Cond: (realm_id = 2)
         Filter: ((recipient_id = 119) AND (sender_id = 5) AND (upper(content) ~~ '%`APPS`, `PROFILE`, `THEME`, `CHANNELS`, `TOPICS`, `MESSAGE FORMATTING`, `KEYBOARD SHORTCUTS`, `HELP`.%'::text))
 Planning Time: 0.821 ms
 Execution Time: 0.113 ms
(6 rows)
```

Probably @timabbott should run this query for CZO and then we can update the index getting used accordingly.

Reply:
is there a er diagram/schema for zulip?

Reply:
I don't think so.

Can you post the above index related query in #backend thread on CZO for faster response ?

Reply:
We use `zerver_message_realm_sender_recipient` as [confirmed by Tim on CZO](https://chat.zulip.org/#narrow/stream/3-backend/topic/query.20message.20sent.20by.20welcome.20bot/near/1851702)